shakenfist/baseobject.py

```diff
@@ -78,31 +78,28 @@ def get_lock_attr(self, name, op):
     # Properties common to all objects which are routed to attributes
     @property
     def state(self):
-        return self._db_get_attribute('state')
+        db_data = self._db_get_attribute('state')
+        if not db_data:
+            return State(None, 0)
+        return State(**db_data)
 
     def update_state(self, state, error_message=None):
         with self.get_lock_attr('state', 'State update'):
-            dbstate = self.state
-            orig_state = dbstate.get('state')
+            orig = self.state
 
             # Ensure state change is valid
-            if state not in self.state_targets[orig_state]:
+            if state not in self.state_targets[orig.value]:
                 raise exceptions.InvalidStateException(
                     'Invalid state change from %s to %s for object %s',
-                    orig_state, state, self.object_type)
+                    orig.value, state, self.object_type)
 
-            dbstate['state'] = state
-            dbstate['state_updated'] = time.time()
-
-            if error_message:
-                dbstate['error_message'] = error_message
-
-            self._db_set_attribute('state', dbstate)
-            self.add_event('state changed', '%s -> %s' % (orig_state, state))
+            new = State(state, time.time(), error_message)
```

Comment:
I feel like new isn't the best name for a variable ever, even if its not a reserved word in python.

Reply:
Improved.